Cabal/Distribution/PackageDescription/Parsec.hs

```diff
@@ -297,7 +300,18 @@ parseGenericPackageDescription' lexWarnings fs = do
                 _ -> do
                     parseFailure pos $ "Invalid source-repository kind " ++ show args
                     pure RepoHead
-            sr <- parseFields sourceRepoFieldDescrs warnUnrec (emptySourceRepo kind) fields
+
+            let (fs, ss) = partitionFields fields
+            -- traverse_ (traverse_ warnSubsection) ss
+
+            sr <- runFieldFancy fs $ SourceRepo kind -- todo "pure fields"
+                <$> affineField    "type"                repoType
+                <*> affineFieldAla "location" FreeText   repoLocation
+                <*> affineFieldAla "module"   Token      repoModule
+                <*> affineFieldAla "branch"   Token      repoBranch
+                <*> affineFieldAla "tag"      Token      repoTag
+                <*> affineFieldAla "subdir"   FilePathNT repoSubdir
```

Comment:
the applicative value will live somewhere else, as we'll need it for pretty-printer